packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -194,21 +194,19 @@ export class InternalModelManager {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
-  async findOne(id: string, throwOnEmptyData: true): Promise<GadgetRecord<RecordShape>>;
-  async findOne(id: string, throwOnEmptyData: false): Promise<GadgetRecord<RecordShape> | null>;
-  async findOne(id: string, throwOnEmptyData = true): Promise<GadgetRecord<RecordShape> | null> {
```

Comment:
I guess maybe we got the overloads wrong. Perhaps to deal with an optional arg one needs to actually do something like:

```
async findOne(id: string): Promise<GadgetRecord<RecordShape>>;
```

We'd probably also have to something similar on `findOneRunner` to make it work. As much as I'd like to do that, let's worry about that in a future PR.

Reply:
Yeah, I can try that another time — maybe when I'm working on fixing the typing issues of the hooks.